src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Integer.Signed.Default.cs

```diff
@@ -2,47 +2,166 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
 namespace System.Buffers.Text
 {
     /// <summary>
     /// Methods to format common data types as Utf8 strings.
     /// </summary>
     public static partial class Utf8Formatter
     {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool TryFormatInt64Default(long value, Span<byte> buffer, out int bytesWritten)
         {
             if ((ulong)value < 10)
             {
-                if (buffer.Length == 0) goto FalseExit;
-                buffer[0] = (byte)('0' + value);
-                bytesWritten = 1;
+                return TryFormatUInt32SingleDigit((uint)value, buffer, out bytesWritten);
+            }
+#if BIT32
+            if (value <= int.MaxValue && value >= int.MinValue)
+            {
+                return TryFormatInt32MultipleDigits((int)value, buffer, out bytesWritten);
+            }
+            else
+            {
+                if (value <= Utf8Constants.Quintillion && value >= -Utf8Constants.Quintillion)
+                {
+                    return TryFormatInt64LessThanQuintillion(value, buffer, out bytesWritten);
+                }
+                else
+                {
+                    return TryFormatInt64MoreThanQuintillion(value, buffer, out bytesWritten);
+                }
+            }
+#else
+            return TryFormatInt64MultipleDigits(value, buffer, out bytesWritten);
+#endif // BIT32
+        }
+
+        // TODO: Use this instead of TryFormatInt64Default to format numbers less than int.MaxValue for BIT32
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFormatInt32Default(int value, Span<byte> buffer, out int bytesWritten)
+        {
+            if ((uint)value < 10)
+            {
+                return TryFormatUInt32SingleDigit((uint)value, buffer, out bytesWritten);
+            }
+            return TryFormatInt32MultipleDigits(value, buffer, out bytesWritten);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFormatInt32MultipleDigits(int value, Span<byte> buffer, out int bytesWritten)
+        {
+            if (value < 0)
+            {
+                value = -value;
+                int digitCount = FormattingHelpers.CountDigits((uint)value);
+                // WriteDigits does not do bounds checks
+                if (digitCount >= buffer.Length)
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+                buffer[0] = Utf8Constants.Minus;
+                FormattingHelpers.WriteDigits((uint)value, buffer.Slice(1, digitCount));
+                bytesWritten = digitCount + 1;
```

Comment:
`bytesWritten = digitCount + 1;` should be *before* the call to WriteDigits to reduce register pressure.

Reply:
Should I do this everywhere (including places where I don't increment digitCount), like https://github.com/dotnet/corefx/pull/25626/files#diff-35d50d92a59705862a380c2b0c0fc5c9R77?

```C#
    FormattingHelpers.WriteDigits(value, buffer.Slice(0, digitCount));
    bytesWritten = digitCount;
```

Reply:
Yes.

Reply:
Can you explain why it helps with register pressure, is it because after returning from Slice, the register holding digitCount is freed up ?

Reply:
Right. And bytesWritten too (if it is held in the register).